fhir-server/src/main/java/com/ibm/fhir/server/rest/FHIRRestInteractionVisitorReferenceMapping.java

```diff
@@ -92,18 +95,19 @@ public FHIRRestOperationResponse doCreate(int entryIndex, FHIRPersistenceEvent e
             final Resource finalResource = visitor.getResult(); // finalResource immutable
 
             // Try offloading storage of the payload. The offloadResponse will be null if not supported
+            String resourcePayloadKey = UUID.randomUUID().toString();
             int newVersionNumber = Integer.parseInt(finalResource.getMeta().getVersionId().getValue());
-            Future<PayloadKey> offloadResponse = storePayload(finalResource, finalResource.getId(), newVersionNumber);
+            PayloadPersistenceResponse actualOffloadResponse = storePayload(finalResource, finalResource.getId(), newVersionNumber, resourcePayloadKey);
```

Comment:
Robin opened https://github.com/IBM/FHIR/issues/3145 so that this happens in a more logical spot